3_javascript/javascript1/week2/js_w2.js

```diff
@@ -0,0 +1,122 @@
+////////////////////////Flight booking fullname function////////////////////////
+
+/////// step1
+// function getFullname(firstname, surname) {
+//     let fullname = firstname + surname;
+//     return fullname;
+//   }
+  
+//   let fullname1 = getFullname("Meraj ", "Sharifi");
+//   let fullname2 = getFullname("Mina ", "Miri");
+  
+//   console.log("Full Name 1:", fullname1);
+//   console.log("Full Name 2:", fullname2);
+  
+
+/////// step2 - Formal fullname
+
+// function getFullname(firstname, surname, useFormalName) {
+//     if (useFormalName) {return "Lord " + firstname + surname;}
+//     else {return firstname + surname;}
+//   }
+  
+//   let formalFullName = getFullname("Meraj ", "Sharifi", true);
+//   let inFormalFullName = getFullname("Mina ", "Miri", false);
+  
+//   console.log("Formal Full Name:", formalFullName);
+//   console.log("Informal Full Name:", inFormalFullName);
+
+  
+/////// FINAL STEP: WITHOUT NAME AND ADDING FEMALE NAME
+
+  function getFullname(firstname, surname, useFormalName, isFemale) {
+    if (useFormalName) {return "Lord " + firstname + surname}
+    else if (!firstname && !surname) {return "Please provide a valid name.";}
+    else if (isFemale) {return "Lady " + firstname + surname}
+    else {return firstname + surname}
+}
+
+  const formalFullName = getFullname("Meraj ", "Sharifi", true, false);
+  const inFormalFullName = getFullname("Mina ", "Miri", false, false);
+  const formalFullFemaleName = getFullname("Sara ", "Ono", false, true);
+  const missingNames = getFullname("", "", false); // Empty names
+
+  console.log("Formal Full Name:", formalFullName);
+  console.log("Informal Full Name:", inFormalFullName);
+  console.log("Informal Full Female Name:", formalFullFemaleName);
+  console.log("Missing Names:", missingNames);
+
```

Comment:
🎉 Good job taking on this coding challenge!
☠️ there is a tiny bug here, when I run this `getFullname("Mina ", "Miri", true, true);` I get `Lord Mina Miri`, shouldn't it be `Lady Mina Miri`
💪 Beautiful job on handling`(!firstname && !surname)` this case